docs/rfc/block-registration.md

```diff
@@ -0,0 +1,369 @@
+
+_TBD: Describe the resolution behavior by file extension, and consider further extension-specific subtypes (e.g. WPDefinedJSPropertyFile)._
+
+### Save
+
+* Type: string (`WPDefinedPropertyFile`)
+* Optional
+* Localized: No
+* Property: `save`
+
+```json
+{ "save": "my-block-save.js" }
+```
+
+This property is a pointer to a JavaScript file containing the save function of the block type. The save function defines the way in which the different attributes should be combined into the final markup, which is then serialized by Gutenberg into `post_content`.
+
+See the [Edit and Save](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-edit-save/) documentation for more details.
+
+If omitted, the implementation should fall back to one as if it were defined as a function returning `null`, where the expected behavior is to serialize without any inner HTML (a "void block").
+
+### Render Callback
+
+* Type: string (`WPDefinedPropertyFile`)
+* Optional
+* Localized: No
+* Property: `renderCallback`
+
+```json
+{ "renderCallback": "my-block-render-callback.php" }
+```
+
+This is a pointer to a php file returning a render callback php function.  The render callback is function called when the block is rendered on the frontend. It's used to generate the frontend markup dynamically.
+
+See the [dynamic blocks documentation](https://wordpress.org/gutenberg/handbook/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks/) for more details.
+
+### Styles Variations
+
+*   Type: array
+*   Optional
+*   Localized: No
+*   Property: `styleVariations`
+
+```json
+{ 
+    "styleVariations": 
+        [ 
+			{ "name": "default", "label": "Default", "isDefault": true }, 
+			{ "name": "other", "label": "Other" }
+		]
+}
+```
+
+Block styles can be used to provide alternative styles to block. It works by adding a class name to the block's wrapper. Using CSS, a theme developer can target the class name for the style variation if it is selected.
+
+Plugins and Themes can also register [custom block style](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#block-style-variations) for existing blocks.
+
+### Transforms
+
+* Type: string
+* Optional
+* Localized: No
+* Property: `transforms`
+
+```json
+{ "transforms": "my-block-transforms.js" }
+```
+
+This property is a pointer to a JavaScript file containing the save function of the block transforms. The save function defines the way in which the different attributes should be combined into the final markup, which is then serialized by Gutenberg into `post_content`.
+
+See the [Transforms](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#transforms-optional) documentation for more details.
+
+### Deprecated versions
+
+* Type: object[]
+* Optional
+* Localized: No
+* Property: `deprecated`
+
+```json
+{ "deprecated": [ { attributes, save, supports } ] }
```

Comment:
I wondered if another option for the deprecated property could be to link to another previous version of the block.json. 

Creating a deprecation could be copying the current implementation into a subfolder, referencing it, and then making changes to the block in the current folder.

Reply:
It seems like a good option but would require the plugin author to keep a lot more code around. I'm not against though.

Reply:
This example needs to be further adjusted as it is not valid JSON at the moment.